src/pages/HappDetails/HappDetails.test.js

```diff
@@ -0,0 +1,178 @@
+import React from 'react'
+import { render, fireEvent, act } from '@testing-library/react'
+import { Router } from 'react-router-dom'
+import { createMemoryHistory } from 'history'
+import { MockedProvider } from '@apollo/react-testing'
+import wait from 'waait'
+import HappDetails from './HappDetails'
+import { appOne as appHoloFuel } from 'mock-dnas/happStore'
+import AllAvailableHappsQuery from 'graphql/AllAvailableHappsQuery.gql'
+import EnableHappMutation from 'graphql/EnableHappMutation.gql'
+import DisableHappMutation from 'graphql/DisableHappMutation.gql'
+
+const enableHappMock = {
+  request: {
+    query: EnableHappMutation,
+    variables: { appId: 'QmHHAHappEntryAddressHash2' }
+  },
+  result: {
+    data: { enableHapp: { id: 'not', title: 'used', happStoreId: 'at', isEnabled: 'all' } }
+  },
+  newData: jest.fn()
+}
+
+const disableHappMock = {
+  request: {
+    query: DisableHappMutation,
+    variables: { appId: 'QmHHAHappEntryAddressHash1' }
+  },
+  result: {
+    data: { disableHapp: { id: 'not', title: 'used', happStoreId: 'at', isEnabled: 'all' } }
+  },
+  newData: jest.fn()
+}
+
+const mocks = [
+  {
+    request: {
+      query: AllAvailableHappsQuery
+    },
+    result: {
+      data: {
+        allAvailableHapps: [
+          {
+            id: 'QmHHAHappEntryAddressHash1',
+            title: 'HoloFuel',
+            description: 'Manage and redeem payments for hosting',
+            dnaHash: 'foiyuoiyZXBVNBVCuibce',
+            happStoreId: 'QmXxiimzfcSHYqHXV2z6WNopeiFnPBx9YKnHzPcq9o8VoT',
+            homepageUrl: 'https://holo.host/faq/what-is-holo-fuel/',
+            isEnabled: true,
+            thumbnailUrl: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcS2cMFvYqaw7TtcTkPFcwE8pupKWqLFMCFu2opap9jqUoqIcAKB',
+            ui: null
+          },
+          {
+            id: 'QmHHAHappEntryAddressHash2',
+            title: 'Holo Community',
+            description: 'Connect with other hosts in the Holo network',
+            dnaHash: 'sd;lmsdl;masd;lmds;lmasdlsadm;ldmo',
+            happStoreId: 'QmXx7imYqHXV2z6WNopeiFnPBx9YKnHzPcq9o8VoTzfcSH',
+            homepageUrl: 'https://hylo.com',
+            isEnabled: false,
+            thumbnailUrl: 'https://d3ngex8q79bk55.cloudfront.net/misc/default_community_avatar.png',
+            ui: null
+          }
+        ]
+      }
+    }
+  },
+  enableHappMock,
+  disableHappMock
+]
+
+function renderWithRouter (
+  ui,
+  {
+    route = '/',
+    history = createMemoryHistory({ initialEntries: [route] })
+  } = {}
+) {
+  return {
+    ...render(<Router history={history}>{ui}</Router>),
+    history
+  }
+}
+
+async function renderHoloFuelApp (address = 'QmHHAHappEntryAddressHash1') {
+  const render = await renderWithRouter(<MockedProvider mocks={mocks} addTypename={false}>
+    <HappDetails history={{}} match={{ params: { address } }} />
```

Comment:
repeating myself, but it's confusing to me to be using `renderWithRouter` here and yet also passing in an empty object for `history`.

Reply:
Not sure if needed here. If not, I'll remove it for clarity.
Thanks!

Reply:
Removed replaced with plain render wherever I can, but history object needs to stay here (unless we want to put this as a prop default)